### DISCUSSION TITLE
How about modules, packages?

### DISCUSSION THREAD

**Discussion by MirKml** [goplus/xgo#1813] [Q&A] [answered]
Using modules, packages are very important for engineering purpose.
How can I use these concepts in goplus?
There is little notice about modules in docs https://github.com/goplus/gop/blob/main/doc/docs.md#module-imports.
But isn't help much. Also mentioned link into "modules" doesn't work.
I can import any go modules, it's clear, but there is no information how can I create, use, own goplus modules and maybe also packages.
How can I structure goplus project with these goplus modules?
What about packages?
How to share modules, packages between projects?

I don't have some previous pure golang experience. Maybe there are many overlaps, reusing some golang tools, it's fine, don't reinvent the wheel. But when it will be in the documentation, it will be very useful.
Thanks.

**Comment by xushiwei** [comment, marked as answer]
Create a new Go+ module is same as Go (suppose module name is `github.com/foo/bar`):

```bash
go mod init github.com/foo/bar  # Or: gop mod init github.com/foo/bar
```

If you import external packages, before compiling you should execute:

```bash
gop mod tidy
```
